docs/rfc/derived-aspects/derived-aspects.md

```diff
@@ -0,0 +1,118 @@
+
+- Value of this aspect gets computed only when `GET` API is invoked. 
+- `GET` API computes, persists and retrieves the metadata from the KV store.
+
+### Entity Aspect Model
+
+Below shows a typical entity - aspect model for derived metadata. 
+
+![Derived Aspect Entity Aspect Model](Entity-Aspect-Model.png)
+ 
+In the above, `DerivedAspect` is an aspect of entity1. The value of this is computed using metadata from 
+- `aspect2` of `entity1`
+- `aspect3` of `entity2`
+- `other metadata` from auxiliary files / other services.
+
+### Computation of Derived Aspect
+ ![Derived Aspect powering index](Powering-Indexes.png)
+
+`GET` API of a typical metadata aspect simply fetches latest value of the aspect from KV store.
+
+Whereas, `GET` API for derived aspect, does something more than that. A typical workflow of `GET` of derived aspect is below.
+ 
+1. Compute the derived metadata
+2. Check if the derived metadata has changed from previous persisted value n KV store.
+3. Persist if the metadata has changed.
+4. Retrieve latest value of derived metadata and serve the `GET` request.
+  
+The persistence of the derived aspect follows GMA architecture. This helps with few of the motivational points mentioned. 
+- Versioning capabilities of the derived metadata 
+- Notification in the form of MAE when derived metadata has changed. 
+
+The MAEs produced will furher power both search and graph GMA indices. This is similar to any other GMA aspect. 
+
+### Refresh of derived aspect metadata
+`GET` operation on derived aspect always gives the current state of derived metadata. 
+
+Often, `GET` operation might not be called for all entity values. Hence, peeking into the DB doesn't guarantee to give the true state of derived aspect.
+   
+Depending on how fresh we need GMA indexes and how often we want to notify on the changes of derived aspect metadata, there is a need for a CRON job which calls GET operation for all URNs of the entity.  
+
+ ![Backfill Derived Aspects](Backfill-Process.png)
+
+In the above, a CRON job on a nightly basis calls `GET` API of derived aspect. This ensures a few things   
+   - KV store is fresh after every 24 hours
+   - Notification of change in derived metadata happens at-least once in 24 hours. 
+       - Note that some other clients can call `GET` API with in 24 hours and there is a possibility of emitting MAE more than once in a day.
+
+## How we teach this
+
+- Derived metadata is derived from multiple aspects of possibly different(or same) entities with(or without) any other auxiliary information.  
+- Derived metadata can be modelled as aspects and are called as `Derived Aspects`. These are very similar to any other GMA aspect.
+- `GET` API of `Derived Aspect` always gives the up-to date derived metadata.
+- A CRON job maintains the freshness of the `Derived Aspect` persisted in KV store and GMA indices.
+
+## Drawbacks
+ 
+As more and more derived aspects are onboarded, additional load on the KV store is inevitable.  
```

Comment:
Also, just noting, but a positive of the kv store is audit history. Can help debug if/when there were bad values and why.

Reply:
Agree. And it is actually a motivation for this RFC 

> 2. Maintain versions of the derived metadata